specification/flag-evaluation/hooks.md

```diff
@@ -2,4 +2,174 @@
 
 ## Overview
 
-// TODO: this is a stub.
+Hooks are a mechanism whereby application developers can add arbitrary behavior
+to flag evaluation. They operate similarly to middleware in many web frameworks.
+
+### Definitions
+
+**Hook**: Application author/integrator-supplied logic that is called by the OpenFeature framework at a specific stage.
+**Stage**: An explicit portion of the flag evaluation lifecycle. e.g. `before` being "before the evaluation is run.
+**Invocation**: A single call to evaluate a flag. `client.getBooleanValue(..)` is an invocation.
+**API**: The global API singleton.
+
+### Hook context
+
+Hook context exists to provide hooks with information about the invocation.
+
+###### Requirement 1.1
+
+> Hook context **MUST** provide: the flag key, evaluation context, default value and a list of  executed hooks by stage.
+
+
+##### Condition 1.2
+
+> If the language type system differentiates between strings, numbers, booleans, and structures.
+>
+>##### Condition 1.2.1
+>
+> Condition: You **MUST** provide `flag type`.
```

Comment:
I think we decide to remove this condition, and just make this a requirement.

Also, avoid use of "you". Use imperative mood / passive voice:

"`Flag type` must be provided" or "`Hook context must contain the flag type`"